www/profile/index.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+$servername = "localhost";
+$username = "worker";
+$password = "swiffcat";
+$db = "socialmedia";
+
+// Create connection
+$conn = new mysqli($servername, $username, $password, $db);
+
+// Check connection
+if (!$conn) {
+    die("Connection failed: " . mysqli_connect_error());
+}
+
+$headers = getallheaders();
+if (!array_key_exists("token", $headers)) {
+    header("HTTP/1.1 401 Unauthorized");
+    die("{\"reason\":\"token not included\"}");
+}
+$token = getallheaders()["token"];
+
+// If invalid header token, die
+$stmt = $conn->prepare("SELECT user_id FROM auth WHERE token = ?");
+$stmt->bind_param("s", $a1);
+$a1 = $token;
+$stmt->execute();
+$stmt->store_result();
+$user_id = null;
+$stmt->bind_result($user_id);
+if ($stmt->fetch()) {
+    // user_id is bound
+} else {
+    header("HTTP/1.1 401 Unauthorized");
+    die();
+}
+
+if ($_SERVER['REQUEST_METHOD'] === 'GET') {
+    if (!isset($_GET["id"])) {
+        header("HTTP/1.1 403 Forbidden");
+        die("{\"reason\":\"no user id\"}");
+    }
+
+    $user_id = $_GET["id"];
+
+    // make sure the user exists
+    $stmt = $conn->prepare("SELECT user_id FROM users WHERE user_id = ?");
```

Comment:
If the user is in auth it exists o_O Why make another query here like this?

Reply:
Because I wanted it to be flexible enough to grab anyone's profile, not just that user's. Like if I wanted to look at someone elses profile.